daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -294,7 +294,7 @@ private[lf] object Speedy {
     /** Compute a stack trace from the locations in the continuation stack.
         The last seen location will come last. */
     def stackTrace(): ImmArray[Location] = {
-      val s = ImmArray.newBuilder[Location]
+      val s = ImmArray.newBuilder[Location](kontStack.size())
```

Comment:
The number of locations in the kontinuation stack will be significantly smaller than the stack itself. Since we don't produce stack traces in a hot loop, I'd suggest we fall back to the array doubling strategy here:
```suggestion
      val s = ImmArray.newBuilder[Location]()
```